src/models/state/wallet/wallet_state.rs

```diff
@@ -466,34 +473,42 @@ impl WalletState {
                     Duration::from_millis(new_block.kernel.header.timestamp.value()),
                     new_block.kernel.header.height,
                 ));
-                monitored_utxos.push(mutxo);
+                monitored_utxos.push(mutxo).await;
             }
 
             // Update mutator set to bring it to the correct state for the next call to batch-update
             msa_state.add(addition_record);
         }
 
         // sanity check
-        let mutxo_with_valid_mps = monitored_utxos
-            .iter()
-            .filter(|(_i, mutxo)| {
-                mutxo.is_synced_to(new_block.kernel.header.prev_block_digest)
-                    || mutxo.blockhash_to_membership_proof.is_empty()
-            })
-            .count();
-        assert_eq!(
-            mutxo_with_valid_mps,
-            valid_membership_proofs_and_own_utxo_count.len(),
-            "Monitored UTXO count must match number of managed membership proofs"
-        );
+        {
+            let stream = monitored_utxos.stream_values().await;
+            pin_mut!(stream); // needed for iteration
+
+            let mutxo_with_valid_mps = stream
```

Comment:
Consider renaming this variable to something that reflects that it counts something.

Reply:
I think it is ok for now.  the variable is only used on 3 subsequent lines, so no real chance of confusion.  might revisit later.